src/hocon_schema.erl

```diff
@@ -84,13 +87,17 @@
          , desc => iodata()
          }.
 
+-type struct_meta() :: #{desc => iodata()}.
 -type field() :: {name(), typefunc() | field_schema()}.
+-type fields() :: [field()] | #{fields := [field()],
```

Comment:
There are more meta's to be added than just `desc`, such as: 
I need to set this meta to mark this configuration as hot updatable, generating the HTTP API put method.